EnhancedMediaPlayer/MediaPlayerTappableView.swift

```diff
@@ -0,0 +1,132 @@
+// MediaPlayerTappableView.swift
+
+import SwiftUI
+
+struct MediaPlayerTappableView: View {
+    @ObservedObject var viewModel: ViewModel
+
+    var body: some View {
+        ZStack {
+            renderSeekLayer()
+            renderControlsLayer()
+        }
+    }
+
+    @ViewBuilder func renderControlsLayer() -> some View {
+        Rectangle()
+            .foregroundColor(.clear)
+            .contentShape(Rectangle())
+            .frame(width: Constants.tappableControlScreenWidth, height: .infinity)
+            .onTapGesture {
+                viewModel.onTapControlsArea?()
+            }
+    }
+
+    @ViewBuilder func renderSeekLayer() -> some View {
+        HStack(spacing: .zero) {
+            renderTappableScreen(.backward)
+            renderTappableScreen(.forward)
+        }
+    }
+
+    @ViewBuilder private func renderTappableScreen(_ seek: Seek) -> some View {
+        ZStack {
+            renderSeekTapsDetector(seek)
+            renderSeekIcon(seek)
+        }
+    }
+
+    @ViewBuilder func renderSeekTapsDetector(_ seek: Seek) -> some View {
+        Rectangle()
+            .foregroundColor(.clear)
+            .contentShape(Rectangle())
+            .onTapGesture {
+                viewModel.onTapSeekArea?(seek)
+            }
+    }
+
+    @ViewBuilder func renderSeekIcon(_ seek: Seek) -> some View {
+        if viewModel.visibleSeek == seek {
+            seek.systemImage
+                .resizable()
+                .frame(width: Constants.iconSize.width, height: Constants.iconSize.height)
+                .foregroundColor(.white)
+        }
+    }
+}
+
+extension MediaPlayerTappableView {
+    class ViewModel: ObservableObject {
+        @Binding var seekState: SeekState
+        @Binding var isSeekByTapEnabled: Bool
+        @Binding var visibleSeek: Seek?
```

Comment:
I don't think you need this `visibleSeek` variable. Isn't this just the same as `seekState.seek`? Having two variables for this means that they can be out of sync, which isn't desireable